portal/home2.php

```diff
@@ -0,0 +1,323 @@
+                'dropdownID' => 'accounting',
+                'children' => [
+                    [
+                        'url' => "#ledgercard",
+                        'label' => xlt('Ledger'),
+                        'icon' => 'fa-folder-open',
+                        'dataToggle' => 'collapse'
+                    ]
+                ]
+            ]);
+        }
+    }
+
+    // Build sub nav items
+
+    if ($GLOBALS['allow_portal_chat']) {
+        array_push(
+            $navItems,
+            [
+                'url' => '#messagescard',
+                'label' => xlt('Chat'),
+                'icon' => 'fa-comment-medical',
+                'dataToggle' => 'collapse',
+                'dataType' => 'cardgroup'
+            ]
+        );
+    }
+
+    for ($i = 0; $i < count($navItems); $i++) {
+        if ($GLOBALS['allow_portal_appointments'] && $navItems[$i]['label'] === text($result['fname'] . " " . $result['lname'])) {
+            array_push($navItems[$i]['children'], [
+                'url' => '#appointmentcard',
+                'label' => xlt("My Appointments"),
+                'icon' => 'fa-calendar-check',
+                'dataToggle' => 'collapse'
+            ]);
+        }
+
+        if ($navItems[$i]['label'] === text($result['fname'] . " " . $result['lname'])) {
+            array_push(
+                $navItems[$i]['children'],
+                [
+                    'url' => 'javascript:changeCredentials(event)',
+                    'label' => xlt('Change Credentials'),
+                    'icon' => 'fa-cog fa-fw',
+                ],
+                [
+                    'url' => 'logout.php',
+                    'label' => xlt('Logout'),
+                    'icon' => 'fa-ban fa-fw',
+                ]
+            );
+        }
+
+        if (!empty($GLOBALS['portal_onsite_document_download']) && $navItems[$i]['label'] === xlt('Reports')) {
+            array_push(
+                $navItems[$i]['children'],
+                [
+                    'url' => '#reportcard',
+                    'label' => xlt('Report Content'),
+                    'icon' => 'fa-folder-open',
+                    'dataToggle' => 'collapse'
+                ],
+                [
+                    'url' => '#downloadcard',
+                    'label' => xlt('Download Lab Documents'),
+                    'icon' => 'fa-download',
+                    'dataToggle' => 'collapse'
+                ]
+            );
+        }
+        if ($GLOBALS['portal_two_payments'] && $navItems[$i]['label'] === xlt('Accountings')) {
+
+            array_push($navItems[$i]['children'], [
+                'url' => "#paymentcard",
+                'label' => xlt('Make Payment'),
+                'icon' => 'fa-credit-card',
+                'dataToggle' => 'collapse'
+            ]);
+        }
+    }
+
+    return $navItems;
+}
+
+$navMenu = buildNav($newcnt, $pid, $result);
+
+echo $twig->render('home.html.twig', [
```

Comment:
would use the standard $twig class that is created in globals.php:
https://github.com/openemr/openemr/blob/master/interface/globals.php#L284

So above line should look like:
```
echo $GLOBALS['twig']->render('portal/home.html.twig', [
```

Then you will have access to some standard functions for escaping and translations in the view layer:
https://github.com/openemr/openemr/blob/master/src/Core/TwigExtension.php

And then the lines 53-59 above should be able to be removed

Reply:
@benmarte , as with all things related to php pain, this appears to be escaping issues (even when i get the path straightened out, turning off autoescaping is breaking something). Playing around with this and should have more details on this soon.